docker_test.go

```diff
@@ -137,10 +143,87 @@ func runPushTest(containerManager container.ContainerManagerType, imageName, mod
 	inttestutils.ContainerTestCleanup(t, serverDetails, artHttpDetails, imageName, tests.DockerBuildName, repo)
 }
 
+func TestRunPushFatManifestImage(t *testing.T) {
+	// Create new container that includes docker daemon + buildx CLI tool
+	// The builder image name.
+	builderImageName := "buildx_builder:1"
+	// The container name which runs the builder image.
+	builderContainerName := "buildx_container"
+	// The image name to build for multi platforms.
+	multiArchImageName := tests.DockerImageName + "-multiarch-image"
+	// The multi platforms image tag .
+	multiArchImageTag := ":latest"
+	buildName := "push-fat-manifest" + tests.DockerBuildName
+	// A name for buildx output file
+	buildxOutputFile := "buildmetadata"
+
+	// Setup test env.
+	workspace, err := filepath.Abs(tests.Out)
+	assert.NoError(t, err)
+	// Build the builder image locally.
+	builderImageName, err = inttestutils.BuildTestImage(builderImageName, "Dockerfile.Buildx.Fatmanifest", container.DockerClient)
+	assert.NoError(t, err)
+	defer inttestutils.DeleteTestImage(t, builderImageName, container.DockerClient)
+
+	// Run the builder container.
+	runCmd := inttestutils.NewRunDockerImage(container.DockerClient, "-d", "--name", builderContainerName, "--privileged", "-v", workspace+":/workspace", builderImageName)
+	assert.NoError(t, gofrogcmd.RunCmd(runCmd))
+	defer inttestutils.DeleteTestcontainer(t, builderContainerName, container.DockerClient)
+
+	// login to the Artifactory within the container
+	password := *tests.JfrogPassword
+	if *tests.JfrogAccessToken != "" {
+		password = *tests.JfrogAccessToken
+	}
+	execCmd := inttestutils.NewExecDockerImage(container.DockerClient, builderContainerName, "docker", "login", *tests.DockerRepoDomain, "--username", *tests.JfrogUser, "--password", password)
+	err = gofrogcmd.RunCmd(execCmd)
+	require.NoError(t, err, "fail to login to container registry")
+
+	// Build & push the multi platform image to Artifactory
+	execCmd = inttestutils.NewExecDockerImage(container.DockerClient, builderContainerName, "/buildx", "build", "--platform", "linux/amd64,linux/arm64,linux/arm/v7", "--tag", path.Join(*tests.DockerRepoDomain, multiArchImageName+multiArchImageTag), "-f", "Dockerfile.Fatmanifest", "--metadata-file", "/workspace/"+buildxOutputFile, "--push", ".")
+	// Docker daemon may be lost for a few seconds, perform 3 retries before failure.
+	for i := 0; i < 3; i++ {
+		if err = gofrogcmd.RunCmd(execCmd); err == nil {
+			break
+		}
+	}
+	require.NoError(t, err, "fail to create and push multi arch image")
```

Comment:
fail to create and push multi arch image
-->
failed creating and pushing multi arch image